common-codegen/src/main/java/io/vrap/rmf/codegen/common/generator/model/immutable/ImmutableModelGenerator.java

```diff
@@ -0,0 +1,227 @@
+            return Stream.empty();
+        }
+        final AnnotationSpec jsonTypeInfoAnnotation = AnnotationSpec.builder(JsonTypeInfo.class)
+                .addMember("use", "$T.NAME", JsonTypeInfo.Id.class)
+                .addMember("include", "$T.PROPERTY", JsonTypeInfo.As.class)
+                .addMember("property", "$S", object.getDiscriminator())
+                .addMember("visible", "true")
+                .build();
+
+        CodeBlock.Builder annotationBodyBuilder = CodeBlock.builder();
+        List<ObjectType> children = getSubtypes(object).collect(Collectors.toList());
+        if (!children.isEmpty()) {
+            for (int i = 0; i < children.size(); i++) {
+                annotationBodyBuilder.add(
+                        (i == 0 ? "{" : ",") +
+                                "\n@$T(value = $T.class, name = $S)"
+                                + (i == children.size() - 1 ? "\n}" : "")
+                        , JsonSubTypes.Type.class, getClassName(getPackagePrefix(), children.get(i)), children.get(i).getDiscriminatorValue());
+            }
+        }
+        final AnnotationSpec jsonSubTypesAnnotation = AnnotationSpec.builder(JsonSubTypes.class)
+                .addMember("value", annotationBodyBuilder.build())
+                .build();
+
+        return Stream.of(jsonSubTypesAnnotation, jsonTypeInfoAnnotation);
+    }
+
+    protected Stream<AnnotationSpec> getAdditionalTypeAnnotations(final ObjectType object) {
+        AnnotationSpec immutablesValueAnnotationSpec = AnnotationSpec.builder(Value.Style.class)
+                .addMember("visibility", "$T.PRIVATE", Value.Style.ImplementationVisibility.class)
+                .addMember("builderVisibility", "$T.PUBLIC", Value.Style.BuilderVisibility.class)
+                .build();
+
+        final ClassName builderName = ClassName.get(getObjectPackage(getPackagePrefix(), object), object.getName() + "Builder");
+        AnnotationSpec jacksonAnnotationSpec = AnnotationSpec.builder(JsonDeserialize.class)
+                .addMember("builder", "$T.class", builderName)
+                .build();
+        return Stream.of(immutablesValueAnnotationSpec, jacksonAnnotationSpec);
+    }
+
+    public Single<GenerationResult> generateStub() {
+
+        Single<GenerationResult> generationResult = getRamlObjects().map(this::transformToJavaFile)
+                .doOnNext(javaFile -> javaFile.writeTo(getOutputFolder()))
+                .map(JavaFile::toJavaFileObject)
+                .toList()
+                .doOnSuccess(list -> CtAnnotationProcessor.processAnnotations(getOutputFolder().toFile(),
+                        Arrays.asList(Nullable.class, Value.Immutable.class, JsonProperty.class, JsonDeserialize.class), list)
+                )
+                .map(javaFiles -> getPaths(javaFiles, getOutputFolder()))
+                .map(GenerationResult::of);
+        return generationResult;
+    }
+
+    public JavaFile transformToJavaFile(final AnyType object) {
```

Comment:
Here we should rename the parameter to `anyType`.

Reply:
nice catch 👍